newdle/api.py

```diff
@@ -77,6 +81,38 @@ def require_token():
     g.user = user
 
 
+@api.route('/avatar/<email>')
```

Comment:
Regarding my suggestion with the signed URLs below: You could use `/avatar/<payload>` and pass both email and initials in the signed string created by the `secure_serializer`.